pywb/static/wombatProxyMode.js

```diff
@@ -0,0 +1,357 @@
+            // We must use the window reference passed to us to access this origins stylesheets
+            var styleSheets = $wbwindow.document.styleSheets;
+            for (; i < styleSheets.length; ++i) {
+                sheet = styleSheets[i];
+                // if the sheet belongs to our parser node we must skip it
+                if (!this.shouldSkipSheet(sheet)) {
+                    try {
+                        // if no error is thrown due to cross origin sheet the urls then just add
+                        // the resolved URLS if any to the media urls array
+                        media = media.concat(this.extractMediaRules(sheet.cssRules, sheet.href || $wbwindow.document.baseURI));
+                    } catch (error) {
+                        // the stylesheet is cross origin and we must re-fetch via PYWB to get the contents for checking
+                        deferredMediaURLS.push(this.corsCSSFetch(sheet.href));
+                    }
+                }
+            }
+            // We must use the window reference passed to us to access this origins elements with srcset attr
+            // like cssRule handling we must include a URL to resolve relative URLs by
+            var srcsetElems = $wbwindow.document.querySelectorAll('*[srcset]');
+            var ssElem, resolveAgainst;
+            for (i = 0; i < srcsetElems.length; i++) {
+                ssElem = srcsetElems[i];
+                resolveAgainst = ssElem.src != null && ssElem.src !== ' ' ? ssElem.src : $wbwindow.document.baseURI;
+                srcset.push({'srcset': ssElem.srcset, 'resolve': resolveAgainst});
+            }
+
+            // send what we have extracted, if anything, to the worker for processing
+            this.postMessage({'type': 'values', 'media': media, 'srcset': srcset});
+
+            if (deferredMediaURLS.length > 0) {
+                // wait for all our deferred fetching and extraction of cross origin
+                // stylesheets to complete and then send those values, if any, to the worker
+                var aaw = this;
+                Promise.all(deferredMediaURLS).then(function (values) {
+                    var results = [];
+                    while (values.length > 0) {
+                        results = results.concat(values.shift());
+                    }
+                    if (results.length > 0) {
+                        aaw.postMessage({'type': 'values', 'media': results});
+                    }
+                });
+            }
+        };
+
+        WBAutoArchivingWorker = new AutoArchivingWorker();
+
+        if (isTop) {
+            $wbwindow.addEventListener("message", function (event) {
+                if (event.data && event.data.wb_type === 'aaworker') {
+                    WBAutoArchivingWorker.postMessage(event.data.msg);
+                }
+            }, false);
+        }
+    }
+
+    if (wbUseAAWorker) {
+        initAutoArchivingWorker();
+    }
+
+    if (wbUseWombat) {
```

Comment:
Hm, do we want to support a mode where there is 'Auto-Archiving' worker but not regular wombat?
Maybe for simplicity, there should just be 3 options:
- No rewriting
- Minimal proxy wombat
- Minimal proxy wombat + auto archiving worker
?